### ISSUE TITLE
Sonarr

### ISSUE THREAD

**Issue by estimadarocha** [theotherp/nzbhydra#238]
Dunno if this a problem with Sonarr or with hydra, but that never happens before i start using nzbhydra.

From times to times Sonarr stop searching from nzbhydra, i discovered if i go to indexers tab on sonnar and if i edit the indexer (nzbhydra) click on test, let it finished and them save get the problem solved.... until it happens again after a while!

**Comment by theotherp** [comment]
Check your sonarr/ hydra log, there was probably an error when hydra tried to search. My last big update screwed some things up and hydra isn't running as smoothly as before. I'm working on it.

**Comment by imadunatic** [comment]
@theotherp , it appears that for whatever reason hydra isn't responding fast enough for Sonarr. I noticed on the 05-11 update that my remote Sonarr server was no longer properly hitting hydra. At first I thought it was a sonarr issue but I started testing it on my local instance and it also fails to contact hydra. 

When I test hydra from sonarr, sonarr tells me it fails and it's due to a gateway timeout response from nginx: 

Sonarr Log
```
NzbDrone.Common.Http.HttpException: HTTP request failed: [504:GatewayTimeout] [GET] at [https://localhost/nzbhydra/api?t=tvsearch&cat=5000&extended=1&apikey=removed&offset=0&limit=100]
   at NzbDrone.Common.Http.HttpClient.Execute(HttpRequest request) in M:\BuildAgent\work\6c3239faf2b92630\src\NzbDrone.Common\Http\HttpClient.cs:line 107
   at NzbDrone.Core.Indexers.HttpIndexerBase`1.FetchIndexerResponse(IndexerRequest request) in M:\BuildAgent\work\6c3239faf2b92630\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:line 285
   at NzbDrone.Core.Indexers.HttpIndexerBase`1.FetchPage(IndexerRequest request, IParseIndexerResponse parser) in M:\BuildAgent\work\6c3239faf2b92630\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:line 271
   at NzbDrone.Core.Indexers.HttpIndexerBase`1.TestConnection() in M:\BuildAgent\work\6c3239faf2b92630\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:line 298
<!DOCTYPE html>
<html>
<head>
<title>Error</title>
<style>
    body {
        width: 35em;
        margin: 0 auto;
        font-family: Tahoma, Verdana, Arial, sans-serif;
    }
</style>
</head>
<body>
<h1>An error occurred.</h1>
<p>Sorry, the page you are looking for is currently unavailable.<br/>
Please try again later.</p>
<p>If you are the system administrator of this resource then you should check
the <a href="http://nginx.org/r/error_log">error log</a> for details.</p>
<p><em>Faithfully yours, nginx.</em></p>
</body>
</html>
```

Oddly, Sonarr is getting through to hydra, but hydra isn't passing anything back quick enough: 
```
2016-05-14 11:08:32,094 - INFO - web - API search request: SearchRequest [ type "tv", category "5000", offset "0", limit "100" ]
2016-05-14 11:08:33,332 - WARNING - search_module - NZB.CAT: Indexer returned a result page without total results and offset. Shame! *rings bell*
```
I tried going back to 0.2.7 to see if Sonarr could at least communicate with Hydra again, but it's still giving me the timeout issue. Is this something you're still working on or should it be working for me now and there is an issue on my end?

**Comment by theotherp** [comment]
Please lower the log level for the log file to debug and let sonar test the connection to hydra. The post the infos from System -> Bugrepreport.

**Comment by theotherp** [comment]
I haven't seen this particular behavior so far, no.

**Comment by imadunatic** [comment]
In troubleshooting I decided to re-issue my api key and reboot, so far that seems to have helped the sonarr issue. Strangely it seemed to be on Sonarr's end, but the logs from Hydra now that it's working is completely different than when it wasn't so I'm thinking this was something specific to my setup. It just seemed weird that after working without issue for months it decided to act up on the 11th, but it just so happened that Sonarr updated that day and I also updated hydra so I lost my control....  I also noticed that I had way more indexers enabled for API access than I needed, so I knocked it down to 2 for API access (and yet again lost my control check). 

I knew better than to sway from the scientific approach but it was one of those days where you just want to get it working and don't pay attention to how you get there. I do know it's working now, but I'll be damned if I know exactly what fixed it.
